client/protocols/unlock.py

```diff
@@ -123,29 +131,26 @@ def interact(self, answer, choices=None, randomize=True):
                 if choices and student_input in choice_map:
                     student_input = choice_map[student_input]
 
-                if not self._verify(student_input, answer[i]):
+                input_lines.append(student_input)
+                if not self._verify(student_input, line):
                     break
-                else:
-                    input_lines.append(student_input)
             else:
                 correct = True
 
-
-            # TODO(albert): record analytis
-            # Performt his before the function exits?
-            # self._analytics[self._analytics['current']].append((attempts, correct))
-
-            # if input_lines.lower() in self.EXIT_INPUTS:
-            #     attempts -= 1
-            #     self._analytics[self._analytics['current']].append((attempts, correct))
-            #     return
+            self.analytics.setdefault(self.current_test, []).append({
+                'id': question_id,
+                'question timestamp': self.unix_time(question_timestamp),
+                'answer timestamp': self.unix_time(datetime.now()),
```

Comment:
I only need the timestamp to be consistent within a student, since they are mostly for time between viewing the questions and the responses. This is good to know though, in case I ever do decide to try using time between students.